File: ClientApp/src/app/models/Post/PostDto.ts
import { CommentDto } from './../Comment/CommentDto';
import { UserDto } from '../User/UserDto';

export class PostDto extends Response{
    public id: string;
    public message: string;
    public likeCount: number;
    public likedByUserIds: Array<string>;
    public owner: UserDto;
    public comments: Array<CommentDto>;
}

File: apigateway/Models/Post/PostDto.cs
using System.Collections.Generic;

namespace apigateway
{
    public class PostDto
    {
        public string Id { get; set; }
        public string Message { get; set; }

        public int LikeCount { get; set; }

        public IEnumerable<string> LikedByUserIds { get; set; }

        public UserDto Owner { get; set; }

        public IEnumerable<CommentDto> Comments { get; set; }
    }
}